src/Misc/SegmentedProgressBar.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { ReactElement, useCallback } from 'react';
+import styled, { keyframes } from 'styled-components';
+
+const TooltipText = styled.span<{width?: string}>`
+  visibility: hidden;
+  padding: 5px;
+  font-size: 10px;
+  color: #000000;
+  line-height: 1.5;
+  text-align: left;
+  border: 1px solid hsl(195deg 5% 60% / 72%);
+  box-shadow: 2px 2px 4px hsl(195deg 16% 72% / 72%);
+  background-color: #ffffff;
+  outline: none !important;
+  position: absolute;
+  top: 222%;
+  left: 12px;
+  z-index: 1;
+  width: ${({ width }) => width ? (parseInt(width) * 1.5)+'px' : '120px'};
+`;
+
+const Container = styled.div<{width?: string}>`
+  width: ${({ width }) => width ? width+'px' : '80px'};
+`;
+
+const ProgressBar = styled.div<{height?: string}>`
+  position: relative;
+  display: flex;
+  align-items: flex-start;
+  flex-wrap: nowrap;
+  background: #ddd;
+  height: ${({height}) => height ? height : '5px'};
+  border-radius: 5px;
+  > div:first-child {
+    border-top-left-radius: 5px;
+    border-bottom-left-radius: 5px;
+  }
+  > div:last-child {
+    border-top-right-radius: 5px;
+    border-bottom-right-radius: 5px
+  }
+`;
+
+const Bar = styled.div`
+  box-sizing: border-box;
+  flex: 1;
+  height: inherit;
+  position: relative;
+  &:hover ${TooltipText} {
+    visibility: visible;
+  }
+`;
+
+const CompletedBar = styled(Bar)`
+  border: 1px solid #4DB5FF;
+  background: #4DB5FF;
+`;
+
+const FailedBar = styled(Bar)`
+  border: 1px solid #EF7878;
+  background: #EF7878;
+  cursor: pointer;
```

Comment:
Why is the Failed bar the only one with pointer?
Maybe add it to all the bars or keep them without since the tooltip is not a clickable element, but a hover one.